Default.sublime-commands

```diff
@@ -141,8 +141,12 @@
         "command": "gs_pull"
     },
     {
-        "caption": "git: pull with rebase",
-        "command": "gs_pull",
+        "caption": "git: pull from branch",
+        "command": "gs_pull_from_branch"
+    },
+    {
+        "caption": "git: pull from branch with rebase",
```

Comment:
Might make sense to also add a `git: pull with rebase` command, as this is the most common use case.

Reply:
The `git: pull` in this PR is aware of `git config pull.rebase`. Maybe there is no need for `git: pull with rebase`?

Do you want to make it explicit that `git: pull` does not use `--rebase` and `git: pull with rebase` uses `--rebase`?

Reply:
Added `git: pull with rebase` which uses "rebase" explicitly while the behaviour of `git: pull` still depends on `git config pull.rebase`.

Reply:
I'm not sure which is best... If we mimic the `git pull` CLI function, defaulting to merge and optionally accepting e.g `--rebase` is the behaviour expected by users.

Now that we also allow override in the config, which is great, but makes one of the commands redundant. But, so does the CLI (user with `pull.rebase` set, won't ever need to use `git pull --rebase`). Those users who default to rebase will need to use I think `git pull --no-rebase` to get the merge behaviour.

In the end, it's about UX.. while I don't think adding commands for all arguments is really a good idea, some common commands shouldn't hurt. I think we should draw the line somewhere; in the end many of the use cases are down to personal (or team) preference. And since all of these commands can definitely be added by users in their local command files, I wonder if we need all of them enabled.

I have a suggestion, how about we add *but comment out* the more exotic cases (such as `git pull with merge`) ? `--pull` is really the only common option I know to `git pull`, even other branches is a rare use case.

Reply:
You are right that `git: pull` and `git: pull with rebase` are redundant when `pull.rebase` is true. But creating the command `git pull with merge` (`--no=rebase`)  just makes the situation more complicated. I call it done now.